hooks/honeybadger/honeybadger.go

```diff
@@ -0,0 +1,36 @@
+package logrus_honeybadger
+
+import (
+	"github.com/DavidHuie/gobadger"
+	"github.com/Sirupsen/logrus"
+)
+
+// HoneybadgerHook to send exceptions to an exception-tracking service compatible
+// with the Honeybadger API.
+type HoneybadgerHook struct {
+	conn *gobadger.Conn
+}
+
+func NewHoneyBadgerHook(apiKey string) *HoneybadgerHook {
+	return &HoneybadgerHook{gobadger.NewConn(apiKey)}
+}
+
+func (hook *HoneybadgerHook) Fire(entry *logrus.Entry) error {
+	err := hook.conn.Error(entry.Message)
```

Comment:
Your idea of implementing slowness protection in logrus core to protect against bad hooks is really not bad. It means we could circuit break it as well.